### DISCUSSION TITLE
Use of Google Cloud service account attached to console instance

### DISCUSSION THREAD

**Discussion by nasmart** [autobase-tech/autobase#749] [General]
Is it possible to use a service account assigned to a Google Cloud VM instance running the console in order to avoiding using a JSON key for the service account?

**Comment by vitabaks** [comment]
Currently, a service account key is required, as per the documentation: https://postgresql-cluster.org/deployment/gcp#prerequisites. However, the key doesn’t need to be provided every time a new cluster is deployed if you save it in the console’s database (are stored in encrypted form) on the Secrets page.

**Comment by nasmart** [comment]
Thanks for the quick response. Unfortunately there is an org policy (`iam.disableServiceAccountKeyCreation`) blocking the creation of keys for service accounts.

**Reply by vitabaks** [reply]
In this case, you can take care of the VM preparation yourself and we can install all the software on them https://postgresql-cluster.org/deployment/your-own-machines
